azkaban-common/src/main/java/azkaban/imagemgmt/dto/RequestContext.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.dto;
+
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This class represents a data transfer object (DTO) to propagate image management REST API json
+ * payload, query params, pagination params etc. This class can be used to pass these information
+ * to the different layers of the API implementation.
+ */
+public class RequestContext {
```

Comment:
Shall it be renamed to ImageManagementRequestContext or something? Only RequestContext doesn't help to figure out the usage of the class.

Reply:
ImageMgmt is already there in the package, so that would duplicate if we prefix this. If just RequestContext is not conveying much about the usage, we can have something like ImageMetadataRequestContext. This would be little long so ImageMetadataRequest should suffice.

Reply:
Sounds good. Thank you for taking care of all the comments.